src/main/java/picard/analysis/GcBiasMetricsCollector.java

```diff
@@ -102,70 +108,111 @@ public PerUnitGcBiasMetricsCollector(final String sample, final String library,
             this.sample = sample;
             this.library = library;
             this.readGroup = readGroup;
-            final String prefix;
-            if (this.readGroup != null) {
-                prefix = this.readGroup;
-                gcData.put(prefix, new GcObject());
-            } else if (this.library != null) {
-                prefix = this.library;
-                gcData.put(prefix, new GcObject());
-            } else if (this.sample != null) {
-                prefix = this.sample;
-                gcData.put(prefix, new GcObject());
-            } else {
-                prefix = allReads;
-                gcData.put(prefix, new GcObject());
+            this.gcData = prepareGcData();
+            if (IS_NON_DUPLICATES) {
+                this.gcDataNonDups = prepareGcData();
             }
         }
 
         /////////////////////////////////////////////////////////////////////////////
         //Takes each record and sends them to addRead to calculate gc metrics for
         // that read for each accumulation level
         /////////////////////////////////////////////////////////////////////////////
+        @Override
         public void acceptRecord(final GcBiasCollectorArgs args) {
             final SAMRecord rec = args.getRec();
-            final String type;
+            if (rec.getReadBases().length == 0) {
+                log.warn("Omitting read " + rec.getReadName() + " with '*' in SEQ field.");
```

Comment:
this can get old very quickly if there are no reads with SEQ bases...limit at 100 warnings please.